components/notebook-controller/Dockerfile

```diff
@@ -7,7 +7,7 @@
 
 # Build arguments
 ARG SOURCE_CODE=.
-ARG GOLANG_VERSION=1.20
+ARG GOLANG_VERSION=1.21
```

Comment:
Thanks for the upgrade, Jiri! Have you had a chance to test and ensure that the controllers work properly in ODH with this upgrade, (potential changes on code syntax etc)? We want to avoid any surprises. Also, do we already have a JIRA ticket for this?

Reply:
It's a bit late trying to get this into 2.11 RHOAI. If we mange to get this to red-hat-data-services main (and the rhoai-2.11 branch) in the Friday merge, that's already code freeze. It may be best to postpone it for (any eventual) 2.11.1 and wait for 2.12. Wdyt? And in that case it's pointless to test it right now.

Reply:
I believe it's valuable to test the functionality before pushing it downstream. It will save time in the future and ensure a smoother integration.

Reply:
Ok, if you're going to help me to get it in, then let's test, sure.

Reply:
Give me a bit more time to take a better look on the testing cluster that you shared!